src/Select.js

```diff
@@ -598,11 +600,24 @@ var Select = React.createClass({
 
 		}, this);
 
-		return ops.length ? ops : (
-			<div className="Select-noresults">
-				{this.props.asyncOptions && !this.state.inputValue ? this.props.searchPromptText : this.props.noResultsText}
-			</div>
-		);
+		if (ops.length) {
+			return ops;
+		} else {
+			var noResultsText = (function () {
+				if (this.state.isLoading) {
+					return this.props.searchingText;
+				} else if (this.state.inputValue || !this.props.asyncOptions) {
+					return this.props.noResultsText;
+				} else {
+					return this.props.searchPromptText;
+				}
+			}.bind(this))();
```

Comment:
nit: Can you use `var this = self` instead of `.bind(this)`
Happy to merge once you squash though.